### ISSUE TITLE
"Open/Save File" Dialog behind main window on Linux

### ISSUE THREAD

**Issue by playgithub** [VSCodium/vscodium#1120]
### Env
* OS: Manjaro
* Desktop: KDE Plasma 5.24.5
* VSCodium: 1.68.0 (BTW the previous version also has the same problem)

### Problem
When opeing or saving a file, if a file name is needed, the popup file dialog is behind the main window instead of in front of the main window.
So the file dialog needs to be bring to front manually everytime.

### More Info
Not tested for vscode, because it's not convenient to test it on Manjaro.

**Comment by jocot** [comment]
confirmed on ubuntu 20.04.4 LTS / Gnome desktop
I get a notification saying "Save As" is ready, and have to alt-tab to the view save file dialog. This behaviour started appearing in the past handful of updates.

Version: 1.69.2
Commit: 3b889b090b5ad5793f524b5d1d39fda662b96a2a
Date: 2022-07-19T00:58:08.152Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Linux x64 5.4.0-122-generic

**Comment by playgithub** [comment]
Still true in 1.70.0

**Comment by javiergcim** [comment]
Still true in 1.71.0

**Comment by daiyam** [comment]
Seems to be an electron bug: https://github.com/electron/electron/issues/32857

**Comment by CarterSnich** [comment]
1.72.2, bug still exists

**Comment by Riddle1001** [comment]
Adding https://extensions.gnome.org/extension/2182/noannoyance/ to get rid of the top pop up notification worked for me.

**Comment by Crystal-RainSlide** [comment]
> 1.72.2, bug still exists on Manjaro KDE 5.25.5. I just installed vscodium. I used vscode from AUR before and the Open/Save file dialog works perfectly fine.

Could be some closed source patch from Microsoft.

**Comment by CarterSnich** [comment]
> Adding https://extensions.gnome.org/extension/2182/noannoyance/ to get rid of the top pop up notification worked for me.

This won't do much help on KDE Plasma, but I found a way to fix this. File dialog now popups over the main codium window and prevents me from click on it.

![image](https://user-images.githubusercontent.com/52433531/200468253-df205a20-a8cb-40ff-a7d0-309f3d49f384.png)

**Comment by mramendi** [comment]
Same problem here. Fedora 36, KDE Plasma 5.26.4, VSCodium 1.74.0

**Comment by weefunker** [comment]
Same problem for months now, especially annoying in vscode 

Ubuntu 20.04.5 LTS

**Comment by nurettin** [comment]
I don't know if this is related, but every time I try saving an unsaved new file, the save-as dialog freezes with 100% CPU usage.

**Comment by mrflos** [comment]
> 



> ![image](https://user-images.githubusercontent.com/52433531/200469421-5c92afa1-e311-4eb7-8fe6-ac290269f61a.png)

This worked for me but I had to use as window class `xdg-desktop-portal-gtk`

**Comment by stotes** [comment]
@mrflos suggestion partially worked for me however it would set all windows above others which isn't the desired result.

The following settings worked better for me running `manjaro.


Window Class: `codium`
Window Type: `dialog`
Window Role [`Exact Match`]: `gtkfilechooserdialog`
---
Add Property...
`Keep above other windows` -> [`Force`] -> `Yes`


---
![image](https://user-images.githubusercontent.com/941055/223574674-fab41a21-e4d9-4e9c-a575-c32fd56e7b77.png)

**Comment by drajathasan** [comment]
> @mrflos suggestion partially worked for me however it would set all windows above others which isn't the desired result.
> 
> The following settings worked better for me running `6.1.12-1-MANJARO Plasma`
> ## Window Class: `codium`
> Window Type: `dialog`
> Window Role [`Exact Match`]: `gtkfilechooserdialog`
> 
> Add Property... `Keep above other windows` -> [`Force`] -> `Yes`
> 
> ![image](https://user-images.githubusercontent.com/941055/223574674-fab41a21-e4d9-4e9c-a575-c32fd56e7b77.png)

Any suggestion for non KDE? I use mate.

**Comment by stotes** [comment]
@drajathasan I've never used mint so I can't say if this will work, however I would suggest you investigate the CCSM "Window Rules" plugin.
http://wiki.compiz.org/WindowMatching#Window_Rules

**Comment by drajathasan** [comment]
@stotes thanks.

**Comment by xjpjose** [comment]
> Mismo problema aquí. Fedora 36, KDE Plasma 5.26.4, VSCodium 1.74.0

solution:
1. Ctrl + ,
2. ventana - nueva ventana
3. new windows dimensions = offset

**Comment by playgithub** [comment]
electron bug: https://github.com/electron/electron/issues/32857

**Comment by securegh** [comment]
Is there any solution to this? I have the same issue here:

Operating System: Zorin OS 16.2
Kernel: Linux 5.15.0-72-generic
Architecture: x86-64

VSCodium:

Version: 1.78.2
Release: 23132
Commit: e06ea6147733c975d26c3d3552462681e05cb147
Date: 2023-05-12T00:42:22.773Z
Electron: 22.3.5
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Linux x64 5.15.0-72-generic
Sandboxed: No

**Comment by BloodyRain2k** [comment]
I just finally switched from Windows to Debian and now I got this issue on my hands...

```
Version: 1.84.2
Release: 23317
Commit: c8ce3ba4bc6b30b3b10edc61481cb85b1d2396bc
Date: 2023-11-13T18:24:10.824Z
Electron: 25.9.2
ElectronBuildId: undefined
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 6.1.0-13-amd64
```

**Comment by daiyam** [comment]
We can't do anything since it's a bug in Electron so it's up to Microsoft to fix it...

**Comment by BloodyRain2k** [comment]
> We can't do anything since it's a bug in Electron so it's up to Microsoft to fix it...

Yeah, that's not gonna get fixed at all then. Why should *they* fix an issue that only exists on a competing OS?

Sour ranting aside, after messing around with whatever options I can find in the OS and what not, I wanna throw an idea into the room.
But since I've literally just switched this week, forgive me if I spout nonsense for a lack of understanding how the whole stuff works:

I noticed that it's possible to set various actions for windows when clicking the titlebar, one of them being "lower" which drops that window behind others.
Since we apparently can't get the handle for the "Save As" dialog, would it be possible to "drop" the main window instead until the dialog is 'done' doing it's dialog things?

It'd be a messy work around, but it could potentially be a workaround for those that can't use KDE's "Window Rules" stuff.
Although I'm really contemplating to switch to KDE now just for *that*.

**Comment by Crystal-RainSlide** [comment]
@BloodyRain2k I got this from a reply in the Electron issue (https://github.com/electron/electron/issues/32857#issuecomment-1697579690):

https://extensions.gnome.org/extension/5410/grand-theft-focus/

---

There are some 3rd party Window management tools that directly work on X or Wayland may help,
if they can be executed just after the file dialog goes behind main window:

- https://old.reddit.com/r/gnome/comments/u929n6/customizable_windows_rules/i5q3v7r/
  - `wmctrl -F -a '<window title of the file dialog>'`
- For Wayland: https://git.sr.ht/~brocellous/wlrctl
  - `wlrctl window focus app_id:codium title:<window title of the file dialog>`

Also, for Compiz, the updated link for its Window Rules feature previously mentioned by stotes:
https://www.compiz-fusion.org/wiki_subdomain/windowmatching.html#Window_Rules

**Comment by kritonios01** [comment]
Still a bug on VS Codium v1.85.1.
I am using Arch (EndeavourOS) on KDE

**Comment by 3xploiton3** [comment]
still a bug on Arch Cinnamon

```
Version: 1.85.1
Release: 23348
Commit: 08e6c15293922dd53a864bb041be381322fee401
Date: 2023-12-14T20:36:42.296Z
Electron: 25.9.7
ElectronBuildId: undefined
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 6.6.9-arch1-1
```

**Comment by driscollw** [comment]
I guess I'll add to the chain. I'm seeing this as well. 

Arch (EndeavourOS) on KDE
Version: 1.85.2
Release: 24019
Commit: c8a36a69377856369f139f39ee56dd3838130fbf
Date: 2024-01-19T18:30:44.948Z
Electron: 25.9.7
ElectronBuildId: undefined
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 6.7.3-arch1-2

**Comment by Jacalz** [comment]
I am also seeing this on Fedora Workstation 39 (Gnome desktop). Quite annoying. VSCodium from Flathub seems to be using the file pickers from the XDG Desktop Portal and those do not seem to be having the issue (they also look a lot nicer). Can the regular VSCodium binaries not be updated to also use the desktop portals for opening files (instead of the old electron provided GTK+3 file pickers)?

I have attached images of the old vs new file pickers below for reference. I hope this helps to solve the issue.
![image](https://github.com/VSCodium/vscodium/assets/25466657/928806df-4066-4a95-a677-8edf97427786)
![image](https://github.com/VSCodium/vscodium/assets/25466657/8350346d-21c2-4afc-b709-307be5a4dba4)

**Comment by Jacalz** [comment]
Why close this? It is obviously an issue for a lot of people

**Comment by DrakaSAN** [comment]
Even if it is an upstream bug, it looks like there are solutions that could work, like [this comment](https://github.com/VSCodium/vscodium/issues/1120#issuecomment-1997553734)

I however agree it would be better to fix it upstream and as such do the issue tracking upstream, but then it would be nice to point that out when closing the issue

For reference and for those that want to press the issue upstream, the upstream issue is [here](https://github.com/microsoft/vscode/issues/146422)

**Comment by crhy** [comment]
I'm having the same issue.  Linux Mint 21.3 Mate with Compiz.  Most frequently annoying with Chromium and Kdenlive.

**Comment by ahmet-cetinkaya** [comment]
I solved in KDE Plasma X11 6.0.5.
![Screenshot_20240729_152410](https://github.com/user-attachments/assets/1276da3b-40c3-4c3b-b422-2a5c52577930)
